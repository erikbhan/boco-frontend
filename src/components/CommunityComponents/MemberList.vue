<template>
  <div v-if="loading">LASTER...</div>
  <ul v-else>
    <li v-for="member in members" :key="member.userId">
      <UserListItemCard :buttons="buttons" :user="member" />
    </li>
  </ul>
</template>

<script>
import UserListItemCard from "@/components/UserProfileComponents/UserListItemCard.vue";
import CommunityService from "@/services/community.service";

export default {
  name: "MemberList",
  components: {
    UserListItemCard,
  },
  props: {
    buttons: Array,
  },
  data() {
    return {
      members: [],
      loading: false,
    };
  },
  methods: {
    async load() {},
  },
  async created() {
    this.loading = true;
    this.members = await CommunityService.getCommunityMembers(
      this.$route.params.communityID
    );
    this.loading = false;
  },
};
</script>
