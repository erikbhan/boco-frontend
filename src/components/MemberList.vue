<template>
  <ul>
    <li v-for="member in memberlist" :key="member.userId">
      <user-list-item-card :admin="admin" :user="member" />
    </li>
  </ul>
</template>

<script>
import UserListItemCard from "./UserProfileComponents/UserListItemCard.vue";

export default {
  data() {
    return {
      memberlist: [
        {
          userId: 2,
          firstName: "erik",
          lastName: "hansen",
        },
        {
          userId: 1,
          firstName: "Test",
          lastName: "Testesen",
        },
      ],
    };
  },
  components: {
    UserListItemCard,
  },
  props: {
    admin: Boolean,
  },
};
</script>
