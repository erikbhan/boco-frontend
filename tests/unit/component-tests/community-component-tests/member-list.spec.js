import { mount } from "@vue/test-utils";
import MemberList from "@/components/CommunityComponents/MemberList.vue"

describe("MemberList component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(MemberList);
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});