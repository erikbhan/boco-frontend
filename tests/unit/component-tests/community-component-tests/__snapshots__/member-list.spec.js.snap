// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberList component renders correctly 1`] = `
<ul>
  
  <li>
    <div
      class="bg-white shadow dark:bg-gray-800 select-none cursor-pointer hover:bg-gray-50 flex items-center p-4"
    >
      <div
        class="h-10 w-10 flex flex-col justify-center items-center mr-4"
      >
        <router-link
          to="/profile/2"
        >
          <img
            alt="profil"
            src=""
          />
        </router-link>
      </div>
      <div
        class="flex-1 pl-1"
      >
        <div
          class="font-medium dark:text-white"
        >
          erik hansen
        </div>
      </div>
      <div
        class="hidden md:block flex-auto"
      >
        <ul
          class="flex justify-center"
        >
          <li>
            <p
              class="ml-2 text-sm font-medium text-gray-500 dark:text-gray-400"
            >
              Gjennomsnitts rating:  
            </p>
          </li>
          
          <li>
            <svg
              class="w-5 h-5 text-gray-300 dark:text-gray-500"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
              />
            </svg>
          </li>
          <li>
            <svg
              class="w-5 h-5 text-gray-300 dark:text-gray-500"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
              />
            </svg>
          </li>
          <li>
            <svg
              class="w-5 h-5 text-gray-300 dark:text-gray-500"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
              />
            </svg>
          </li>
          <li>
            <svg
              class="w-5 h-5 text-gray-300 dark:text-gray-500"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
              />
            </svg>
          </li>
          <li>
            <svg
              class="w-5 h-5 text-gray-300 dark:text-gray-500"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
              />
            </svg>
          </li>
          
          <li>
            <p
              class="ml-2 text-sm font-medium text-gray-500 dark:text-gray-400"
            >
              "[object Promise]" out of 5 
            </p>
          </li>
        </ul>
      </div>
      <div
        class="flex flex-row justify-center"
      >
        <button
          class="px-4 py-2 font-medium tracking-wide text-white capitalize transition-colors duration-200 transform bg-blue-600 rounded-md hover:bg-blue-500 focus:outline-none focus:ring focus:ring-blue-300 focus:ring-opacity-80"
        >
           Åpne chat 
        </button>
        <!--v-if-->
      </div>
    </div>
  </li>
  <li>
    <div
      class="bg-white shadow dark:bg-gray-800 select-none cursor-pointer hover:bg-gray-50 flex items-center p-4"
    >
      <div
        class="h-10 w-10 flex flex-col justify-center items-center mr-4"
      >
        <router-link
          to="/profile/1"
        >
          <img
            alt="profil"
            src=""
          />
        </router-link>
      </div>
      <div
        class="flex-1 pl-1"
      >
        <div
          class="font-medium dark:text-white"
        >
          Test Testesen
        </div>
      </div>
      <div
        class="hidden md:block flex-auto"
      >
        <ul
          class="flex justify-center"
        >
          <li>
            <p
              class="ml-2 text-sm font-medium text-gray-500 dark:text-gray-400"
            >
              Gjennomsnitts rating:  
            </p>
          </li>
          
          <li>
            <svg
              class="w-5 h-5 text-gray-300 dark:text-gray-500"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
              />
            </svg>
          </li>
          <li>
            <svg
              class="w-5 h-5 text-gray-300 dark:text-gray-500"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
              />
            </svg>
          </li>
          <li>
            <svg
              class="w-5 h-5 text-gray-300 dark:text-gray-500"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
              />
            </svg>
          </li>
          <li>
            <svg
              class="w-5 h-5 text-gray-300 dark:text-gray-500"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
              />
            </svg>
          </li>
          <li>
            <svg
              class="w-5 h-5 text-gray-300 dark:text-gray-500"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
              />
            </svg>
          </li>
          
          <li>
            <p
              class="ml-2 text-sm font-medium text-gray-500 dark:text-gray-400"
            >
              "[object Promise]" out of 5 
            </p>
          </li>
        </ul>
      </div>
      <div
        class="flex flex-row justify-center"
      >
        <button
          class="px-4 py-2 font-medium tracking-wide text-white capitalize transition-colors duration-200 transform bg-blue-600 rounded-md hover:bg-blue-500 focus:outline-none focus:ring focus:ring-blue-300 focus:ring-opacity-80"
        >
           Åpne chat 
        </button>
        <!--v-if-->
      </div>
    </div>
  </li>
  
</ul>
`;
